Modules/Loadable/Segmentations/Widgets/qMRMLSegmentationGeometryWidget.cxx

```diff
@@ -0,0 +1,803 @@
+    // to make sure that if the ROI size is set to match the image size exactly then we
+    // output extent contains the whole image
+    double tolerance = 0.001;
+    outputExtent[1] = ceil((farCorner_Segmentation[0] - origin_Segmentation[0]) / outputSpacing[0] + tolerance) - 1;
+    outputExtent[3] = ceil((farCorner_Segmentation[1] - origin_Segmentation[1]) / outputSpacing[1] + tolerance) - 1;
+    outputExtent[5] = ceil((farCorner_Segmentation[2] - origin_Segmentation[2]) / outputSpacing[2] + tolerance) - 1;
+    this->GeometryImageData->SetExtent(outputExtent);
+    }
+
+    return QString();
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLSegmentationGeometryWidgetPrivate::updateGeometryWidgets()
+{
+  int dims[3] = { 0 };
+  this->GeometryImageData->GetDimensions(dims);
+  double dimsDouble[3] = { (double)dims[0], (double)dims[1], (double)dims[2] };
+  this->MRMLCoordinatesWidget_Dimensions->setCoordinates(dimsDouble);
+
+  // Apply inverse source axis permutation
+  double spacing[3] = { 0.0 };
+  this->GeometryImageData->GetSpacing(spacing);
+  double outputSpacing[3]  = {
+    spacing[this->SourceAxisIndexForInputAxis[0]],
+    spacing[this->SourceAxisIndexForInputAxis[1]],
+    spacing[this->SourceAxisIndexForInputAxis[2]] };
+
+  bool blocked = this->MRMLCoordinatesWidget_Spacing->blockSignals(true);
+  this->MRMLCoordinatesWidget_Spacing->setCoordinates(outputSpacing);
+  this->MRMLCoordinatesWidget_Spacing->blockSignals(blocked);
+
+  double origin[3] = { 0.0 };
+  this->GeometryImageData->GetOrigin(origin);
+  this->MRMLCoordinatesWidget_Origin->setCoordinates(origin);
+
+  vtkNew<vtkMatrix4x4> directions;
+  this->GeometryImageData->GetDirectionMatrix(directions);
+  for (int i=0; i<3; ++i)
+    {
+    for (int j=0; j<3; ++j)
+      {
+      this->MatrixWidget_Directions->setValue(i, j, directions->GetElement(i,j));
+      }
+    }
+}
+
+//-----------------------------------------------------------------------------
+void qMRMLSegmentationGeometryWidgetPrivate::resetGeometryImageData()
+{
+  vtkNew<vtkMatrix4x4> identityMatrix;
+  identityMatrix->Identity();
+  this->GeometryImageData->SetImageToWorldMatrix(identityMatrix);
+  this->GeometryImageData->SetDimensions(0, 0, 0);
+}
+
+//-----------------------------------------------------------------------------
+bool qMRMLSegmentationGeometryWidgetPrivate::isSourceASegmentationWithBinaryLabelmapMaster()
+{
+  vtkMRMLTransformableNode* sourceNode = vtkMRMLTransformableNode::SafeDownCast(
+    this->MRMLNodeComboBox_SourceGeometryNode->currentNode() );
+  vtkMRMLSegmentationNode* sourceSegmentationNode = vtkMRMLSegmentationNode::SafeDownCast(sourceNode);
+  std::string binaryLabelmapName = vtkSegmentationConverter::GetBinaryLabelmapRepresentationName();
+  vtkSmartPointer<vtkOrientedImageData> sourceBinaryLabelmap;
+  if ( sourceSegmentationNode
+    && sourceSegmentationNode->GetSegmentation()->GetNumberOfSegments() > 0
```

Comment:
sourceSegmentationNode->GetSegmentation() can be nullptr, so it would be safer to check that